Add example code for cross-namespace certificateRefs (#14288)
hashicorp/consul: website/content/docs/api-gateway/configuration/gateway.mdx

```

> #### Example cross-namespace certificateRef

Author: This section follows the pattern established by the corresponding cross-namespace docs for routes [here](https://www.consul.io/docs/api-gateway/configuration/routes#example-cross-namespace-backendref).

Reviewer: Terrific. Thanks for adding this. I'm going to add a suggestion that, if we think is an improvement, we can apply to the routes docs, too.

> The following example creates a `Gateway` named `example-gateway` in namespace `gateway-namespace`. This `Gateway` has a `certificateRef` in namespace `secret-namespace`. The reference is allowed because the `ReferenceGrant`, named `reference-grant` in namespace `secret-namespace`, allows `Gateways` in `gateway-namespace` to reference `Secrets` in `secret-namespace`.

Reviewer: ```suggestion
The following example creates a `Gateway` named `example-gateway` in namespace `gateway-namespace` (lines 2-4). The gateway has a `certificateRef` in namespace `secret-namespace` (lines 16-18). The reference is allowed because the `ReferenceGrant` configuration, named `reference-grant` in namespace `secret-namespace` (lines 23-26), allows `Gateways` in `gateway-namespace` to reference `Secrets` in `secret-namespace` (lines 30-34).
```

> <CodeBlockConfig filename="gateway_with_referencegrant.yaml">

Reviewer: ```suggestion
<CodeBlockConfig filename="gateway_with_referencegrant.yaml" lineNumbers=true highlight="2-4","16-18","23-26","30-34">
```
I think this is the correct syntax. So we can make the paragraphing introducing the example a little easier to parse.

> The following example creates a `Gateway` named `example-gateway` in namespace `gateway-namespace` (lines 2-4). The gateway has a `certificateRef` in namespace `secret-namespace` (lines 16-18). The reference is allowed because the `ReferenceGrant` configuration, named `reference-grant` in namespace `secret-namespace` (lines 23-26), allows `Gateways` in `gateway-namespace` to reference `Secrets` in `secret-namespace` (lines 30-34).

Reviewer: ```suggestion
The following example creates a `Gateway` named `example-gateway` in namespace `gateway-namespace` (lines 2-4). The gateway has a `certificateRef` in namespace `secret-namespace` (lines 16-18). The reference is allowed because the `ReferenceGrant` configuration, named `reference-grant` in namespace `secret-namespace` (lines 23-26), allows `Gateways` in `gateway-namespace` to reference `Secrets` in `secret-namespace` (lines 30-35).
```

> <CodeBlockConfig filename="gateway_with_referencegrant.yaml" lineNumbers highlight="2-4,16-18,23-26,30-34">

Reviewer: ```suggestion
<CodeBlockConfig filename="gateway_with_referencegrant.yaml" lineNumbers highlight="2-4,16-18,23-26,30-35">
```

> The following example creates a `Gateway` named `example-gateway` in namespace `gateway-namespace` (lines 2-4). The gateway has a `certificateRef` in namespace `secret-namespace` (lines 16-18). The reference is allowed because the `ReferenceGrant` configuration, named `reference-grant` in namespace `secret-namespace` (lines 23-26), allows `Gateways` in `gateway-namespace` to reference `Secrets` in `secret-namespace` (lines 30-35).
> 
> <CodeBlockConfig filename="gateway_with_referencegrant.yaml" lineNumbers highlight="2-4,16-18,23-26,30-35">

Reviewer: ```suggestion
The following example creates a `Gateway` named `example-gateway` in namespace `gateway-namespace` (lines 2-4). The gateway has a `certificateRef` in namespace `secret-namespace` (lines 16-18). The reference is allowed because the `ReferenceGrant` configuration, named `reference-grant` in namespace `secret-namespace` (lines 24-27), allows `Gateways` in `gateway-namespace` to reference `Secrets` in `secret-namespace` (lines 31-35).

<CodeBlockConfig filename="gateway_with_referencegrant.yaml" lineNumbers highlight="2-4,16-18,24-27,31-35">
```
Sorry! I was still off. I need to learn how to count.

> <CodeBlockConfig filename="gateway_with_referencegrant.yaml" lineNumbers=true highlight="2-4","16-18","23-26","30-34">

Author: Incorrect syntax
```suggestion
<CodeBlockConfig filename="gateway_with_referencegrant.yaml" lineNumbers highlight="2-4,16-18,23-26,30-34">
```
